app/Livewire/Storage/Form.php

```
@@ -91,9 +92,24 @@ public function submit()
+            DB::transaction(function () {
+                $this->validate();
+                $this->storage->save();
+
+                // Test connection with new values - if this fails, transaction will rollback
+                $this->storage->testConnection(shouldSave: false);
+
+                // If we get here, the connection test succeeded
+                $this->storage->is_usable = true;
+                $this->storage->unusable_email_sent = false;
+                $this->storage->save();
+            });
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Facades\DB' in method 'submit'. (undefined)

(StaticAccess)